### PR TITLE
#25 implement full archive for chat

### PR DESCRIPTION
**Pull request by amiedavis**
Delivers #25

### CODE REVIEW

File: provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+	@Override
+	public void archive(final String archiveId, final String siteId, final String toolId, final boolean includeStudentContent) {
+
+		List<ChatChannel> chatChannels = chatManager.getContextChannels(siteId, true);
+
+		for (ChatChannel chatChannel : chatChannels) {
+
+			// Save the metadata for this channel
+			ChatChannelMetadata metadata = new ChatChannelMetadata();
```
Comment (steveswinsburg) [inline provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java]: This block could go to new method like the others

File: provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+			messagesToSave.add(simpleChatMessage);
+		}
+		
+		return messagesToSave;
+	}
+
+	/**
+	 * Simplified helper class to represent an individual chat message
+	 */
+	private static class SimpleChatMessage {
```
Comment (steveswinsburg) [inline provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java]: -static

File: provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+		@Getter @Setter
+		private String userId;
+
+	}
+
+	/**
+	 * Simplified helper class to represent the metadata for a chat channel
+	 */
+	private static class ChatChannelMetadata {
```
Comment (steveswinsburg) [inline provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java]: -static

File: provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+	 * @return the list of messages to be saved
+	 */
+	private List<SimpleChatMessage> createArchiveItem(List<ChatMessage> chatMessages) {
+		List<SimpleChatMessage> messagesToSave = new ArrayList<SimpleChatMessage>();
+		for (ChatMessage message : chatMessages) {
+			SimpleChatMessage simpleChatMessage = new SimpleChatMessage();
+			simpleChatMessage.setBody(message.getBody());
+			simpleChatMessage.setDate(message.getMessageDate());
+			try {
+				simpleChatMessage.setOwner(userDirectoryService.getUser(message.getOwner()).getDisplayName());
```
Comment (steveswinsburg) [inline provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java]: These two calls can be condensed to one. And use a method to get the user. If user doesn't exist, use the uuid. Also userId in simple chat message should be eid.

**Review by steveswinsburg — CHANGES_REQUESTED** [review]

File: provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java (hunk 4 above)
Comment (steveswinsburg) [reply provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java]: If you have a list, call it something plural. Then inside you make items.those can be a method on their own that return just one thing.

So createArchiveItems and createArchiveItem

File: provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java (hunk 1 above)
Comment (amiedavis) [reply provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java]: durp

File: provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java (hunk 4 above)
Comment (amiedavis) [reply provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java]: Which two calls can be condensed into one? I'm confused.
Also aren't I already using the eid?

**Review by steveswinsburg — APPROVED** [review]

**Pull request merged**